dht_net.go

```diff
@@ -153,6 +159,9 @@ func (dht *IpfsDHT) sendRequest(ctx context.Context, p peer.ID, pmes *pb.Message
 
 	ms, err := dht.messageSenderForPeer(ctx, p)
 	if err != nil {
+		if err == msmux.ErrNotSupported {
```

Comment:
oooof, this error should be in go-libp2p-core.

Reply:
do we want to close/reset the stream in this case?

Reply:
I think reset is correct

Reply:
oh, since the negotiation failed, it gets reset anyways. no actual stream gets created here

Reply:
👍